src/main/java/io/rpg/model/actions/BattleAction.java

```diff
@@ -0,0 +1,28 @@
+package io.rpg.model.actions;
+
+import io.rpg.model.object.GameObject;
+import io.rpg.model.object.Player;
+
+public class BattleAction {
+    private final Player player;
```

Comment:
You can access player from Controller, there is no need for him in BattleAction. It will be harder to parse BattleAction from config if you leave player there.